src/helpers/questionnaires.js

```diff
@@ -24,13 +25,35 @@ exports.removeCard = async (cardId) => {
   await CardHelper.removeCard(cardId);
 };
 
-exports.getUserQuestionnaires = async (course, credentials) => {
-  const isCourseStarted = get(course, 'slots.length') && DatesHelper.isAfter(Date.now(), course.slots[0].startDate);
-  if (course.format === STRICTLY_E_LEARNING || isCourseStarted) return [];
+exports.findQuestionnaire = async (course, credentials, type) => Questionnaire
+  .findOne({ type, status: PUBLISHED }, { type: 1, name: 1 })
+  .populate({ path: 'histories', match: { course: course._id, user: credentials._id } })
+  .lean({ virtuals: true });
 
-  const questionnaire = await Questionnaire.findOne({ type: EXPECTATIONS, status: PUBLISHED }, { type: 1, name: 1 })
-    .populate({ path: 'histories', match: { course: course._id, user: credentials._id } })
+exports.getUserQuestionnaires = async (courseId, credentials) => {
+  const course = await Course.findOne({ _id: courseId })
+    .populate({ path: 'slots', select: '-__v -createdAt -updatedAt' })
+    .populate({ path: 'slotsToPlan', select: '_id' })
     .lean({ virtuals: true });
 
-  return !questionnaire || questionnaire.histories.length ? [] : [questionnaire];
+  if (course.format === STRICTLY_E_LEARNING) return [];
+
+  const isCourseStarted = get(course, 'slots.length') && DatesHelper.isAfter(Date.now(), course.slots[0].startDate);
+  if (!isCourseStarted) {
+    const questionnaire = await this.findQuestionnaire(course, credentials, EXPECTATIONS);
+
+    return !questionnaire || questionnaire.histories.length ? [] : [questionnaire];
+  }
+
+  if (get(course, 'slotsToPlan.length')) return [];
```

Comment:
Que penses-tu de mettre cette ligne dans le if de isCourseEnded ? Pour bien comprendre qu'il a sa place non pas comme un test indépendant mais bien uniquement si isCourseStarted = true

Reply:
done

Reply:
Il faut juste changer le "should return an empty array if course is started" en 'if course is ended' dans le nom du test associé